index.html

```diff
@@ -846,6 +901,284 @@
                 });
 
                 /*
+                 * @jiteshjha
+                 * addConnectingBlock(@parameters) makes a new block
+                 * with it's related data structure
+                 * NOTE: This function does not create any ports
+                 * Courtesy: Thanks to Chhavi and Nimish for the code and logic
+                 */
+                function addConnectingBlock(graph, x, y, ifaceFuncName, height, width, relative) {
+                  var details_instance = null;
+                  var connectingBlock = null;
+                  if (ifaceFuncName != null) {
+                      details_instance = new window[ifaceFuncName]();
+                  }
+                  if (details_instance != null) {
+                      var details = details_instance.define();
+                      var enc = new mxCodec(mxUtils.createXmlDocument());
+                      var node = enc.encode(details);
+                      var temp = enc.encode(graph.getDefaultParent());
+                      var stylesheet = graph.getStylesheet();
+                      // From the stylesheet, get the style of the particular block
+                      var style = stylesheet.styles[ifaceFuncName];
+                      /*
+                       * When a particular block is loaded for the first time,
+                       * the image in the style of the block will be a path to the image.
+                       * Set the label in the style property of the block has a html image,
+                       * and set the image in the style property as null
+                       *
+                       * NOTE: Since the image of any block need not be changed for
+                       *       for every movement of that block, the image must be
+                       *       set only once.
+                       */
+                      if (style != null && style['image'] != null) {
+                          // Make label as a image html element
+                          var label = '<img src="' + style['image'] + '" height="80" width="80">';
+                          // Set label
+                          style['label'] = label;
+                          //style['imagePath'] = style['image'];
```

Comment:
Is there a reason that you've commented this block?

Also, CMSCOPE stopped working on refresh again. Any comments?